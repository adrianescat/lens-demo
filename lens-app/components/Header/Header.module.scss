.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 60px;
  border-bottom: 1px solid #f0f0f0;
  width: 100%;
}

.brand {
  font-family: 'MSQTSemiBold';
  color: #000;
}