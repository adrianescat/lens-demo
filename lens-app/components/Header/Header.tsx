import React, { ReactElement } from 'react';
import classnames from 'classnames';
import styles from './Header.module.scss';
import Title from '../Title';
import UserMenu from '../UserMenu';
import { User } from '../../types/user';
import Link from 'next/link';

export interface Props {
  user: User;
}

export default function Header({ user }: Props): ReactElement {
  return (
    <header className={classnames(styles.header, 'container')}>
      <Link href="/">
        <>
          <Title className={styles.brand} />
        </>
      </Link>
      <UserMenu user={user} />
    </header>
  )
}
