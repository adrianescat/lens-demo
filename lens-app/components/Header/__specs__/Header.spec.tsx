import React from 'react';
import { shallow } from 'enzyme';
import Header, { Props } from '../Header';

let wrapper: any;
let props: any;

beforeEach(() => {
  props = { user: { id: 'asd', name: 'Adrian Escat' } };
  wrapper = shallow(<Header {...props} />);
});

test('should match snapshots', () => {
  expect(wrapper).toMatchSnapshot();
});