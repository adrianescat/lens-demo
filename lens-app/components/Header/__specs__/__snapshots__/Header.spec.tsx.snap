// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshots 1`] = `
<header
  className="header container"
>
  <Link
    href="/"
  >
    <Title
      className="brand"
    />
  </Link>
  <UserMenu
    user={
      Object {
        "id": "asd",
        "name": "Adrian Escat",
      }
    }
  />
</header>
`;
